### ISSUE TITLE
NoClassDefFoundError: kotlin.collections.EmptySet

### ISSUE THREAD

**Issue by LikeTheSalad** [open-toast/gummy-bears#23]
Hi!

I've just tried to bump up the signature version to the latest one `0.6.0` in [this project](https://github.com/open-telemetry/opentelemetry-android/blob/main/buildSrc/src/main/kotlin/otel.java-library-conventions.gradle.kts#L17) but after doing so I get this error every time I try to run the `animalsnifferMain` task:

```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':auto-instrumentation:okhttp:okhttp-3.0:library:animalsnifferMain'.
Caused by: : java.lang.NoClassDefFoundError: kotlin.collections.EmptySet
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:109)
	at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:541)
	at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:218)
	at org.gradle.internal.metaobject.AbstractDynamicObject.invokeMethod(AbstractDynamicObject.java:163)
	at org.gradle.api.internal.project.antbuilder.AntBuilderDelegate.nodeCompleted(AntBuilderDelegate.java:124)
	at ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer$_run_closure3$_closure11.doCall$original(AnimalSniffer.groovy:172)
	at ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer$_run_closure3$_closure11.doCall(AnimalSniffer.groovy)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer$_run_closure3.doCall$original(AnimalSniffer.groovy:169)
	at ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer$_run_closure3.doCall(AnimalSniffer.groovy)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at org.gradle.util.internal.ClosureBackedAction.execute(ClosureBackedAction.java:73)
	at org.gradle.util.internal.ClosureBackedAction.execute(ClosureBackedAction.java:54)
	at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder.lambda$execute$0(DefaultIsolatedAntBuilder.java:136)
	at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder.lambda$execute$2(DefaultIsolatedAntBuilder.java:158)
	at ...
```

It seems like some kotlin dependencies aren't available for the task to run properly. Any ideas on how it could be solved? I tried to add the kotlin std lib dependency [here](https://github.com/open-telemetry/opentelemetry-android/blob/main/buildSrc/build.gradle.kts#L19) without luck so far.

**Comment by ogolberg** [comment]
Animal Sniffer uses java serialization for signature data. I converted the code that generates the signatures to Kotlin and likely changed the actual type of this set https://github.com/mojohaus/animal-sniffer/blob/master/animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/Clazz.java#L50 in some cases.

I will post a fix shortly.

**Comment by LikeTheSalad** [comment]
Got it, thanks!

**Comment by ogolberg** [comment]
Fixed in 0.6.1

**Comment by LikeTheSalad** [comment]
Cool! I checked and it's working now, cheers.
